### PR TITLE
new escrow

### PR DESCRIPTION
**Pull request by hav-noms**
WIP. Spec and test cases TBC

### CODE REVIEW

File: SIPS/sip-new-escrow.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+1. Only escrows SNX for 12 months.
+2. Only `FeePool` has the authority to create escrow entries.
+3. The escrow table `checkAccountSchedule` only supports returning up to 5 years of escrow entries from the initial inflationary supply.
+4. `Vest` can only be called by the owner of the SNX.
+
+New requirements for General `SynthetixEscrow` contract
+
+1. Ability to add arbitrary escrow periods. e.g. 3 months to 2 years.
+2. Public escrowing. Allows anyone or any to contract escrow SNX. Allowing SNX to be escrowed for protocol contributors, investors and funds or contracts that escrow some sort of incentive similar to the Staking Rewards.
```
Comment (justinjmoses) [inline SIPS/sip-new-escrow.md:40]: [np] any => anyone

File: SIPS/sip-new-escrow.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+<!--The motivation is critical for SIPs that want to change Synthetix. It should clearly explain why the existing protocol specification is inadequate to address the problem that the SIP solves. SIP submissions without sufficient motivation may be rejected outright.-->
+
+Current limitations of the `RewardEscrow` contract
+
+1. Only escrows SNX for 12 months.
+2. Only `FeePool` has the authority to create escrow entries.
+3. The escrow table `checkAccountSchedule` only supports returning up to 5 years of escrow entries from the initial inflationary supply.
+4. `Vest` can only be called by the owner of the SNX.
+
+New requirements for General `SynthetixEscrow` contract
```
Comment (justinjmoses) [inline SIPS/sip-new-escrow.md:37]: this belongs is the specification section

File: SIPS/sip-new-escrow.md (hunk 1 above)
Comment (hav-noms) [reply SIPS/sip-new-escrow.md]: ta, fixed

File: SIPS/sip-new-escrow.md (hunk 2 above)
Comment (hav-noms) [reply SIPS/sip-new-escrow.md]: done

File: SIPS/sip-new-escrow.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+3. The escrow table `checkAccountSchedule` only supports returning up to 5 years of escrow entries from the initial inflationary supply.
+4. `Vest` can only be called by the owner of the SNX.
+
+Desired features for a general `SynthetixEscrow` contract
+
+1. Ability to add arbitrary escrow periods. e.g. 3 months to 2 years.
+2. Public escrowing. Allows any EOA or any contract to escrow SNX. Allowing SNX to be escrowed for protocol contributors, investors and funds or contracts that escrow some sort of incentive similar to the Staking Rewards.
+3. Update `checkAccountSchedule` to allow for terminal inflation and an unlimited escrow navigation through paging.
+4. Allowing anyone to `vest` an accounts escrowed tokens allows Synthetix network keepers to help support SNX holders and supports the [Liquidation system](sips.synthetix.io/sips/sip-59) to vest an under collateralised accounts vest-able SNX to be paid to the liquidator.
+
```
Comment (jacko125) [inline SIPS/sip-new-escrow.md:44]: Another desired feature for liquidations would be to allow escrowed tokens to be transferred to another account

Comment (hav-noms) [reply SIPS/sip-new-escrow.md:44]: thanks yep I forgot that feature

**Pull request merged**